pagopa/io-react-native-integrity
# feat: [SIW-910] Add iOS methods with example app

### Short description

This PR adds iOS implementation for `DCAppAttestService`. This PR does not add verification steps for attestation or assertion but only the main native swift code and its counterpart on rn bridge.

https://github.com/pagopa/io-react-native-integrity/assets/49342144/67d9df17-550a-4465-8155-c218185e21cf

#### List of Changes

- Add Swift code related to the following methods:
  - isAttestationServiceAvailable
  - generateHardwareKey
  - getAttestation
  - generateHardwareSignatureWithAssertion
- Update example app

#### How Has This Been Tested?

_Integrity check can be tested only on physical device with a valid team and bundle identifier_

1. run **yarn** on the root project folder to install npm dependencies
2. cd example/ios && pod install && cd..
3. yarn example start
4. open xcode to the example/ios/IoReactNativeIntegrityExample.xcworkspace
5. connect a supported iOS device
